### DISCUSSION TITLE
Update `rye` to the latest commit (rather than to the latest official release)

### DISCUSSION THREAD

**Discussion by ma-sadeghi** [astral-sh/rye#860] [Q&A] [answered]
Is there a way to update `rye` to the latest commit (rather than to the latest official release)?

What I currently do is to repeat the "compile yourself" step everytime I need to update to the latest commit. It'd be nice if there was something like `rye self update latest` for instance.

**Comment by bluss** [comment, marked as answer]
Try rye self update, see docs, it looks like it can do it. https://rye-up.com/guide/commands/self/update/

Edit: using --branch main  and previously --rev refs/heads/main

**Reply by ma-sadeghi** [reply]
Oops, I missed the `--rev` option. Thanks!

**Reply by mitsuhiko** [reply]
Unfortunately there is no `--branch`. That would need to be added to point to latest.

**Reply by ma-sadeghi** [reply]
Right, would you consider adding it? Not that it's super important though, the rev works just fine

**Reply by mitsuhiko** [reply]
Absolutely. If you have time add a PR otherwise this goes to the backlog :)

**Reply by ischaojie** [reply]
I'd like to help for this.

**Reply by bluss** [reply]
I thought rev could take a name like a branch, sorry, then it's not really a solution until that's fixed.

I think --rev refs/heads/main will actually work

**Reply by ischaojie** [reply]
It seems cargo is solved like this (cargo install), what is used behind `rye self update`:
```
      --git <URL>             Git URL to install the specified crate from
      --branch <BRANCH>       Branch to use when installing from git
      --tag <TAG>             Tag to use when installing from git
      --rev <SHA>             Specific commit to use when installing from git
```
